Assets/MixedRealityToolkit/Inspectors/Setup/BuildDeployWindow.cs

```diff
@@ -202,268 +217,201 @@ public static void OpenWindow()
-                                }
+            EditorUserBuildSettings.wsaSubtarget = (WSASubtarget)EditorGUILayout.Popup("Target Device", (int)EditorUserBuildSettings.wsaSubtarget, deviceNames, GUILayout.Width(HALF_WIDTH));
 
-                                // To prevent potential confusion, only show this when C# projects will be generated
-                                if (EditorUserBuildSettings.wsaGenerateReferenceProjects &&
-                                    PlayerSettings.GetScriptingBackend(BuildTargetGroup.WSA) == ScriptingImplementation.WinRTDotNET)
-                                {
-                                    // Allow unsafe code
-                                    bool curAllowUnsafeCode = UwpBuildDeployPreferences.AllowUnsafeCode;
-                                    bool newAllowUnsafeCode = EditorGUILayout.ToggleLeft(allowUnsafeCode, curAllowUnsafeCode);
+#if !UNITY_2019_1_OR_NEWER
+            var curScriptingBackend = PlayerSettings.GetScriptingBackend(BuildTargetGroup.WSA);
+            if (curScriptingBackend == ScriptingImplementation.WinRTDotNET)
+            {
+                EditorGUILayout.HelpBox(".NET Scripting backend is deprecated in Unity 2018 and is removed in Unity 2019.", MessageType.Warning);
+            }
 
-                                    if (newAllowUnsafeCode != curAllowUnsafeCode)
-                                    {
-                                        UwpBuildDeployPreferences.AllowUnsafeCode = newAllowUnsafeCode;
-                                    }
-                                }
-                            }
-                        }
+            var newScriptingBackend = (ScriptingImplementation)EditorGUILayout.IntPopup("Scripting Backend", (int)curScriptingBackend, scriptingBackendNames, scriptingBackendEnum, GUILayout.Width(HALF_WIDTH));
+            if (newScriptingBackend != curScriptingBackend)
+            {
+                //bool canUpdate = !Directory.Exists(BuildDeployPreferences.AbsoluteBuildDirectory);
+                // TODO: Troy - still needed? isn't Unity smart enough?
+                /*
+                if (EditorUtility.DisplayDialog("Attention!",
+                        $"Build path contains project built with {newScriptingBackend.ToString()} scripting backend, while current project is using {curScriptingBackend.ToString()} scripting backend.\n\nSwitching to a new scripting backend requires us to delete all the data currently in your build folder and rebuild the Unity Player!",
+                        "Okay", "Cancel"))
+                {
+                    Directory.Delete(BuildDeployPreferences.AbsoluteBuildDirectory, true);
+                    //canUpdate = true;
+                }*/
 
-                        // Generate C# Project References for debugging	
-                        if (PlayerSettings.GetScriptingBackend(BuildTargetGroup.WSA) == ScriptingImplementation.WinRTDotNET)
```

Comment:
Looks like your merge removed some recent changes in mrtk_dev

Reply:
@keveleigh , hold on until I publish the PR. I just wanted to draft it as FYI it's coming. I'll re-request review when I convert from draft

Reply:
@keveleigh, yes it is being broken up into multiple PRs. See item #6520 for the additional issues to come. 

This change mainly involves refactoring such as breaking some of the monolithic functions into sub-components and creating re-usable functions for redundant code, etc. Because of that, it'll be hard to compare diffs. I find it's a lot easier looking at the final result of file(s) (i.e the BuildDeployWindow.cs in this case) alone and reading that it makes sense.